src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019 Datadog, Inc.
+
+using System;
+using System.Threading.Tasks;
+using System.Text;
+using Serilog.Debugging;
+using System.Net.Http;
+using Serilog.Events;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Serilog.Sinks.Datadog.Logs
+{
+    public class DatadogHttpClient : IDatadogClient
+    {
+        private readonly DatadogConfiguration _config;
+        private readonly string _url;
+        private const string _content = "application/json";
+        private readonly LogFormatter _formatter;
+        private HttpClient _client;
+        private readonly Encoding utf8 = Encoding.UTF8;
+        private const int _maxSize = 2 * 1024 * 1024 - 51;  // Need to reserve space for at most 49 "," and "[" + "]"
+        private const int _maxMessageSize = 256 * 1024;
+
+        /// <summary>
+        /// Max number of retries when sending failed.
+        /// </summary>
+        private const int MaxRetries = 1;
+
+        /// <summary>
+        /// Max backoff used when sending failed.
+        /// </summary>
+        private const int MaxBackoff = 30;
+
+        /// <summary>
+        /// Shared UTF8 encoder.
+        /// </summary>
+        private static readonly UTF8Encoding UTF8 = new UTF8Encoding();
+
+        public DatadogHttpClient(DatadogConfiguration config, LogFormatter formatter, string apiKey)
+        {
+            _config = config;
+            _client = new HttpClient();
+            _url = $"{config.Url}/v1/input/{apiKey}";
+            _formatter = formatter;
+        }
+
+        public async Task WriteAsync(IEnumerable<LogEvent> events)
+        {
+            var chunks = SerializeEvents(events);
+            var tasks = chunks.Select(Post);
+
+            await Task.WhenAll(tasks).ConfigureAwait(false);
```

Comment:
What does the false do exactly ?

Reply:
It prevents our Sink awaits from interfering with our clients applications